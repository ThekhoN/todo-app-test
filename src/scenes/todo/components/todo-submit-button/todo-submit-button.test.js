import React from "react";
import { shallow } from "enzyme";
import AddTodoButton from "./index";

describe("AddTodoButton", () => {
  let wrapper, onClick;
  beforeEach(() => {
    onClick = jest.fn();
    wrapper = shallow(<AddTodoButton onClick={onClick} />);
  });
  it("snapshots correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("runs onClick on click", () => {
    wrapper.simulate("click");
    expect(onClick.mock.calls.length).toEqual(1);
    // same as
    // expect(onClick).toBeCalled();
  });
});
