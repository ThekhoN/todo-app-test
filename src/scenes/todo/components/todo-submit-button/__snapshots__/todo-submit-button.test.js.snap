// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodoButton snapshots correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddToDoButton
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Add Todo",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Add Todo",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Add Todo",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Add Todo",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
